### PR TITLE
Add ClearArea method

### PR DESCRIPTION
**Pull request by dizzib**
Smoketest works but might need refinement.

### CODE REVIEW

File: examples/smoketest/cleararea.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+var x11 = require('../../lib');
+var Window = require('./wndwrap');
+
+x11.createClient(function(err, display) {
+    var X = display.client;
+    var w = new Window(X, 0, 0, 700, 500);
+    w.map();
+
+    var render = function() {
+      //X.ChangeGC(w.gc.id, { foreground: w.black, background: w.white });
```
Comment (santigimeno) [inline examples/smoketest/cleararea.js]: remove commented line

**Comment by santigimeno** [comment]
LGTM

**Comment by sidorares** [comment]
btw ChangeGC is currently broken. I fixed it locally yesterday and push today

**Comment by sidorares** [comment]
@santigimeno what do you think would be the best way to test 'visual' requests? Draw something on pixmap, GetImage and compare pixels with expectation?

**Comment by sidorares** [comment]
@dizzib - I think smoketest fails without timeout because you need mapped window for ClearArea. You might move code to Expose event handler

**Comment by santigimeno** [comment]
@sidorares sounds good. Apparently it's what `X Test Suite` does. See: http://cgit.freedesktop.org/xorg/test/xts/tree/xts5/src/lib/checkarea.c#n134, and http://cgit.freedesktop.org/xorg/test/xts/tree/xts5/Xlib9/XClearArea.m#n111

**Comment by santigimeno** [comment]
@sidorares as a side node, porting the `X Test Suite` would be great (I know it's too much work and too little time :) )

**Comment by sidorares** [comment]
@dizzib @santigimeno it turned out a little bit more difficult to write test as ClearArea only accepts windows ( not any drawable / prixmaps ). I'll merge this now and try to add test later

**Pull request merged**